src/ceph-disk/ceph_disk/main.py

```diff
@@ -2123,20 +2179,89 @@ def __init__(self, args):
         super(PrepareBluestoreBlock, self).__init__(args)
 
     def get_space_size(self):
-        return 0  # get as much space as possible
+        block_size = get_conf(
+            cluster = self.args.cluster,
+            variable = 'bluestore_block_size',
+        )
 
-    def desired_partition_number(self):
-        if self.args.block == self.args.data:
-            num = 2
+        if block_size is None:
+            return 0  # get as much space as possible
         else:
-            num = 0
-        return num
+            return int(block_size) / 1048576 # MB
+
+    def desired_partition_number(self):
+        # FIXME: use a fixed partition number for now
```

Comment:
Thanks for throwing out these questions and I agree with your opinion. From my view, we can't always predict user cases. One OSD's block{.wal,.db} could be on separated device, e.g, on SSD, even colocated with other OSD's block{.wal,.db}, while OSD's data and block is still on HDD. Or all the partitions of one OSD are on one SSD device. Actually we are testing both OSD partition layouts now for bluestore to compare the performance and cost.

So you are right and I also think we should set num = 0 for block{,.wal,.db} partition to let it pick first available partition. Do you agree? If yes, I will update accordingly.   

Thanks.